modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -123,42 +122,143 @@
+     * from within {@link TableManager#dropTableAsync(String)}
+     */
+    private final Map<IgniteUuid, CompletableFuture<Void>> dropTblIntention = new ConcurrentHashMap<>();
 
     /**
      * Creates a new table manager.
      *
-     * @param nodeCfgMgr Node configuration manager.
      * @param clusterCfgMgr Cluster configuration manager.
-     * @param metaStorageMgr Meta storage manager.
-     * @param schemaMgr Schema manager.
-     * @param affMgr Affinity manager.
      * @param raftMgr Raft manager.
+     * @param baselineMgr Baseline manager.
+     * @param metaStorageMgr Meta storage manager.
      * @param partitionsStoreDir Partitions store directory.
      */
     public TableManager(
-        ConfigurationManager nodeCfgMgr,
         ConfigurationManager clusterCfgMgr,
-        MetaStorageManager metaStorageMgr,
-        SchemaManager schemaMgr,
-        AffinityManager affMgr,
         Loza raftMgr,
+        BaselineManager baselineMgr,
+        MetaStorageManager metaStorageMgr,
         Path partitionsStoreDir
     ) {
-        this.nodeCfgMgr = nodeCfgMgr;
         this.clusterCfgMgr = clusterCfgMgr;
-        this.metaStorageMgr = metaStorageMgr;
-        this.affMgr = affMgr;
         this.raftMgr = raftMgr;
-        this.schemaMgr = schemaMgr;
+        this.baselineMgr = baselineMgr;
+        this.metaStorageMgr = metaStorageMgr;
         this.partitionsStoreDir = partitionsStoreDir;
     }
 
     /** {@inheritDoc} */
     @Override public void start() {
-        //TODO: IGNITE-14652 Change a metastorage update in listeners to multi-invoke
-        clusterCfgMgr.configurationRegistry().getConfiguration(TablesConfiguration.KEY).tables().listen(ctx -> {
-            return onConfigurationChanged(ctx.storageRevision(), ctx.oldValue(), ctx.newValue());
+        clusterCfgMgr.configurationRegistry().getConfiguration(TablesConfiguration.KEY).tables().
+            listenElements(new ConfigurationNamedListListener<TableView>() {
+            @Override
+            public @NotNull CompletableFuture<?> onCreate(@NotNull ConfigurationNotificationEvent<TableView> ctx) {
+                // Empty assignments might be a valid case if tables are created from within cluster init HOCON
+                // configuration, which is not supported now.
+                assert ((ExtendedTableView)ctx.newValue()).assignments() != null :
+                    "Table =[" + ctx.newValue().name() + "] has empty assignments.";
+
+                final IgniteUuid tblId = IgniteUuid.fromString(((ExtendedTableView)ctx.newValue()).id());
+
+                // TODO: IGNITE-15409 Listener with any placeholder should be used instead.
+                ((ExtendedTableConfiguration) clusterCfgMgr.configurationRegistry().
+                    getConfiguration(TablesConfiguration.KEY).tables().get(ctx.newValue().name())).schemas().
+                    listenElements(new ConfigurationNamedListListener<>() {
```

Comment:
If you subscribe to the concrete table, you have to remove this subscription on drop.

Reply:
Nope, configuration subscriptions will be automatically removed on proper cfg node removal.